publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -452,10 +452,14 @@ private void validatePublication(Publication publicationUpdate, Customer custome
         }
     }
 
-    private void autoCompletePendingPublishingRequestsIfNeeded(Publication publication) {
+    private void autoCompletePendingPublishingRequestsIfNeeded(Publication publication, RequestInfo requestInfo) {
         fetchPendingPublishingRequest(publication)
             .map(PublishingRequestCase.class::cast)
-            .forEach(ticket -> ticket.complete(publication, getOwner(publication)).persistUpdate(ticketService));
+            .forEach(ticket -> ticket.complete(publication, getUserName(requestInfo)).persistUpdate(ticketService));
```

Comment:
getUserName() actually polls conginto each time you call it. Instead do it once outside the forEach

Reply:
Sending userInstance instead, thank you :)